packages/template-retail-react-app/app/pages/product-detail/index.test.js

```diff
@@ -31,15 +32,215 @@ const MockedComponent = () => {
     )
 }
 
-// Set up and clean up
-beforeAll(() => {
-    // Since we're testing some navigation logic, we are using a simple Router
-    // around our component. We need to initialize the default route/path here.
-    window.history.pushState({}, 'ProductDetail', '/en-GB/product/test-product')
-})
+const MockedPageWithProductSet = () => {
+    return (
+        <Switch>
+            <Route
+                path="/en-GB/product/:productId"
+                render={(props) => <ProductDetail {...props} product={mockedProductSet} />}
+            />
+        </Switch>
+    )
+}
+
+const basketWithProductSet = {
```

Comment:
Mocking this data inside this test file. I'm not sure what our approach is. Sometimes the mocked data is in separate file, sometimes not. I guess it depends on whether the data is going to be reused?

Reply:
I agree, and I would add one more consideration: sometimes I separate the mock data just because it obscures the test and makes it harder to think about if the mock data is too long.

I would vote to move this out to a `<test-name>.mock.js` file and have it live in the same directory (similar naming convention keeps them adjacent in your IDE) just because it distracts from the test